### PR TITLE
Catalog Backend: AWS Organizations SDK version bump

### PR DESCRIPTION
**Pull request by ayshiff**
## Hey, I just made a Pull Request!

Related to #3953

This PR removes the aws sdk v2 in `plugins/catalog-backend` and install the new AWS service package for Amazon S3 `@aws-sdk/client-s3`.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/catalog-backend/src/ingestion/processors/AwsOrganizationCloudAccountProcessor.test.ts [hunk 1]
```diff
@@ -73,23 +69,19 @@ describe('AwsOrganizationCloudAccountProcessor', () => {
       const location = { type: 'aws-cloud-accounts', target: 'o-1vl18kc5a3' };
       listAccounts.mockImplementation(() => {
         return {
```
Comment (backjo) [inline plugins/catalog-backend/src/ingestion/processors/AwsOrganizationCloudAccountProcessor.test.ts:71]: I think this mock needs to return a promise

File: yarn.lock [hunk 2]
```diff
@@ -2261,9 +2436,11 @@
     to-fast-properties "^2.0.0"
 
 "@backstage/catalog-model@^0.2.0":
-  version "0.6.0"
+  version "0.2.0"
```
Comment (freben) [inline yarn.lock:2439]: these @backstage changes here and below seem accidental and should be removed

Comment (OrkoHunter) [reply yarn.lock:2439]: They appear frequently when doing `yarn add` or `yarn remove`! Yet to create that issue for more discussion on this problem/hack for Roadie plugins.

@ayshiff You could manually revert these `@backstage` changes. I sometimes copy over the `yarn.lock` file from the master branch and running a fresh `yarn`.

**Comment by backjo** [comment]
Changes look good on the organizations processor - though I think the PR title should be updated to 'AWS Organizations SDK version bump' instead of S3

**Review by freben — APPROVED** [review]

**Pull request merged**